src/compile/data/optimizers.ts

```diff
@@ -43,12 +43,18 @@ export function moveParseUp(node: DataFlowNode) {
     }
 
     if (parent.numChildren() > 1) {
+      // don't move parse further up but continue with parent.
       return true;
     }
 
     if (parent instanceof ParseNode) {
       parent.merge(node);
     } else {
+      // don't swap with nodes that produce something that the parse node depends on (e.g. lookup)
```

Comment:
Why "that the parse node depends on (e.g. lookup)"?  

There is no indication that this is a parse node?